plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergParquetFileWriter.java

```diff
@@ -34,12 +35,12 @@
 import static java.util.Objects.requireNonNull;
 import static org.apache.iceberg.parquet.ParquetUtil.fileMetrics;
 
-public class IcebergParquetFileWriter
-        extends ParquetFileWriter
+public final class IcebergParquetFileWriter
```

Comment:
> Avoid using inheritance to implement IcebergParquetFileWriter

maybe put some rationale in the commit body

Reply:
added commit message